### PR TITLE
Wallet connect & sign arbitrary

### PR DESCRIPTION
**Pull request by jigar-arc10**
- Wallet Connect
- cosmos-kit integration
- merge current praetor functionality to new wallet connect
- store jwt for process API calls
- public key storage for encrypting things in future

### CODE REVIEW

File: apps/provider-console/src/context/SettingsProvider/SettingsProviderContext.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+const defaultSettings: Settings = {
+  apiEndpoint: "",
+  rpcEndpoint: "",
+  isCustomNode: false,
+  nodes: [],
+  selectedNode: null,
+  customNode: null
+};
+
+export const SettingsProvider: FC<{ children: ReactNode }> = ({ children }) => {
```
Comment (baktun14) [inline apps/provider-console/src/context/SettingsProvider/SettingsProviderContext.tsx:54]: I'm not sure you guys need all the logic for custom nodes and list of nodes, but I guess it works without a ui to change it.

File: apps/provider-console/src/types/deployment.ts [hunk 2]
```diff
@@ -0,0 +1,335 @@
+export interface DeploymentDetail {
```
Comment (baktun14) [inline apps/provider-console/src/types/deployment.ts:1]: We should extract all the types into a separate package

**Review by baktun14 — APPROVED** [review]
There's a lot of code duplication with deploy Console. We should try and extract what can be re-used between both apps into the ui package or for type a new types package.

**Comment by jigar-arc10** [comment]
> There's a lot of code duplication with deploy Console. We should try and extract what can be re-used between both apps into the ui package or for type a new types package.

Let's plan the next phase for extraction and shared packages

**Pull request merged**